### PR TITLE
fix: C4 dataloading

### PR DESCRIPTION
**Pull request by timurcarstensen**
I noticed that there is an issue with how the C4 dataset is setup and loaded, this caused/causes crashes since we need to call `tokenizer.encode` as we are dealing with a litgpt tokenizer. Also, we don't return a valid PyTorch `Dataloader` for the validation set. Could you take a look at the code in `data.py` @aaronkl ? :)

TODOs
- [x] fix tokenizer call
- [x] fix val dataloader return

### CODE REVIEW

**Comment by timurcarstensen** [comment]
@aaronkl I removed the C4 dataset entirely and made some changes to the tests. I had to change some floats in the tests, not so sure if that is perhaps an issue. Could you check? :)

File: test/test_pruners.py [hunk 1]
```diff
@@ -77,6 +81,6 @@ def test_model_pruning(model_info, mock_tokenizer):
         nsamples=32,
     )
 
-    assert abs(sparsity_ratio_magnitude - 0.5) <= 0.1
```
Comment (aaronkl) [inline test/test_pruners.py:80]: any idea why this number changes?

File: whittle/pruning/pruners/base_pruner.py [hunk 2]
```diff
@@ -126,7 +126,11 @@ def _prepare_calibration_input(
                 with torch.amp.autocast(device_type="cuda", dtype=torch.bfloat16):
                     with torch.no_grad():
                         torch.cuda.empty_cache()
-                        model(batch[0].to(dev))
+                        if isinstance(batch, (tuple, list)):
```
Comment (aaronkl) [inline whittle/pruning/pruners/base_pruner.py:129]: Is this only needed for the unit tests?

Comment (timurcarstensen) [reply whittle/pruning/pruners/base_pruner.py:129]: No, for any dataloader where the batch is a `dict`. I encountered this with a couple of the datasets that we use. I think this actually isn't a problem with the dataset in the tests but more so with the new default (Alpaca)

File: test/test_pruners.py (hunk 1 above)
Comment (timurcarstensen) [reply test/test_pruners.py:80]: Absolutely not :D
